streaming/java/streaming-api/src/main/java/org/ray/streaming/api/stream/DataStream.java

```diff
@@ -124,19 +147,32 @@ public DataStream(DataStream input, StreamOperator streamOperator) {
    * @return This stream.
    */
   public DataStream<T> partitionBy(Partition<T> partition) {
-    this.partition = partition;
-    return this;
+    checkPartitionCall();
+    return setPartition(partition);
   }
 
   /**
-   * Set parallelism to current transformation.
-   *
-   * @param parallelism The parallelism to set.
-   * @return This stream.
+   * If parent stream is a python stream, we can't call partition related methods
+   * in the java stream.
+   */
+  private void checkPartitionCall() {
+    if (getInputStream() != null && getInputStream().getLanguage() == Language.PYTHON) {
+      throw new RuntimeException("Partition related methods can't be called on a " +
+          "java stream if parent stream is a python stream.");
+    }
+  }
+
+  /**
+   * Convert this stream as a python stream.
+   * The converted stream and this stream are the same logical stream, which has same stream id.
+   * Changes in converted stream will be reflected in this stream and vice versa.
    */
-  public DataStream<T> setParallelism(int parallelism) {
-    this.parallelism = parallelism;
-    return this;
+  public PythonDataStream asPython() {
```

Comment:
toPythonStream() sounds better?

Reply:
The returned stream and previous stream are actually the same stream, will it be better to  name it as `asPythonStream` or `toPythonStream `?